hfile.c

```diff
@@ -506,12 +518,13 @@ static off_t fd_seek(hFILE *fpv, off_t offset, int whence)
 
 static int fd_flush(hFILE *fpv)
 {
-    hFILE_fd *fp = (hFILE_fd *) fpv;
-    int ret;
+    int ret = 0;
     do {
 #ifdef HAVE_FDATASYNC
+        hFILE_fd *fp = (hFILE_fd *) fpv;
         ret = fdatasync(fp->fd);
-#else
+#elif defined(HAVE_FSYNC)
+        hFILE_fd *fp = (hFILE_fd *) fpv;
         ret = fsync(fp->fd);
```

Comment:
Less invasive to add a `#elif defined HAVE_FLUSHFILEBUFFERS`-based version (or add a `FlushFileBuffers()`-based `fsync` function elsewhere for the existing `fd_flush` code to use).